src/Http/Http/src/StreamPipeWriter.cs

```diff
@@ -265,7 +265,7 @@ private void AddSegment(int sizeHint = 0)
             }
 
             // Get a new buffer using the minimum segment size, unless the size hint is larger than a single segment.
-            _currentSegmentOwner = _pool.Rent(Math.Max(_minimumSegmentSize, sizeHint));
+            _currentSegmentOwner = _pool.Rent(Math.Min(Math.Max(_minimumSegmentSize, sizeHint), _pool.MaxBufferSize));
```

Comment:
Are there regression tests for these StreamPipeWriter fixes?

Reply:
I'm making separate PRs for the StreamPipeWriter. I'll make a PR shortly for that fix 😄

Reply:
https://github.com/aspnet/AspNetCore/pull/7143